app/src/main/java/org/sagebase/crf/step/CrfClinicDataGroupsStepLayout.java

```diff
@@ -69,21 +67,38 @@ protected void checkForExistingClinicAssignment() {
         if (!(DataProvider.getInstance() instanceof BridgeDataProvider)) {
             throw new IllegalStateException("CrfClinicDataGroupsStepLayout only works with BridgeDataProvider");
         }
+
         String loadingTitle = getString(R.string.crf_data_groups_loading_title);
         showLoadingDialog(loadingTitle);
+
         BridgeDataProvider bridgeDataProvider = (BridgeDataProvider)DataProvider.getInstance();
         final WeakReference<View> weakView = new WeakReference<>(this);
-        bridgeDataProvider.getStudyParticipant().observeOn(AndroidSchedulers.mainThread()).subscribe(participant -> {
+        List<String> localDataGroups = bridgeDataProvider.getLocalDataGroups();
+
+        bridgeDataProvider.getStudyParticipant().observeOn(AndroidSchedulers.mainThread())
+                .subscribe(participant -> {
             hideLoadingDialog();
             // Controls canceling an observable perform through weak reference to the view
             if (weakView.get() == null || weakView.get().getContext() == null) {
                 return; // no callback
             }
-            List<String> dataGroups = participant.getDataGroups();
-            if (dataGroups.contains(CrfDataProvider.CLINIC1) ||
-                    dataGroups.contains(CrfDataProvider.CLINIC2)) {
+            List<String> serverDataGroups = participant.getDataGroups();
+
+            boolean shouldSkip = true;
+            if (!localDataGroups.contains(CrfDataProvider.TEST_USER)) {
+                if (serverDataGroups.contains(CrfDataProvider.CLINIC1)) {
+                    bridgeDataProvider.addLocalDataGroup(CrfDataProvider.CLINIC1);
```

Comment:
Why would this class be the one adding local data groups?  Wouldn't the response to getStudyParticipant() in BridgeDataProvider update the local data groups?

Reply:
I put it in for now. 

This was handling logic for clearing out old data groups, so if a user signed in again, they could change their groups. Previously, the onboarding flow was additive in the data groups. The ability to change data groups if you repeat onboarding added a lot of complexity here and edge cases from the way Bridge scheduling works. There's a thread in iosdev discussing this, and my first task today is to avoid these issues by skipping data group selection if they already have data

Reply:
Oh yea, I guess a simpler explanation of the code starts with pointing out I'm not adding all the server groups. Just clinic1/clinic2 to prevent them from changing while allowing the user to change the other data groups.